src/Infrastructure/Manifest/Interaction/ManifestInteractionProcessor.php

```diff
@@ -0,0 +1,200 @@
+            default:
+                throw new InvalidArgumentException(sprintf('Invalid mapping in `%s` node', $id));
+        }
+    }
+
+    /**
+     * @param array $value
+     * @param \SplStack $keyStack
+     * @param array $receivedValues
+     *
+     * @return array
+     */
+    protected function processNested(array $value, SplStack $keyStack, array $receivedValues): array
+    {
+        foreach ($value as $itemId => $itemValue) {
+            $keyStack->push($itemId);
+
+            $receivedValues = $this->receiveValue($itemId, $itemValue, $keyStack, $receivedValues);
+
+            $keyStack->pop();
+        }
+
+        return $receivedValues;
+    }
+
+    /**
+     * @param string $id
+     * @param \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Config\ValueCollection $valuesCollection
+     * @param \SplStack $keyStack
+     * @param array $receivedValues
+     *
+     * @return array
+     */
+    protected function processCollection(
+        string $id,
+        ValueCollection $valuesCollection,
+        SplStack $keyStack,
+        array $receivedValues
+    ): array {
+        $index = 0;
+
+        while (
+            ($index === 0 && $valuesCollection->isMinOneItemRequired())
+            || $this->newCollectionItemQuestion->ask($id)
+        ) {
+            $keyStack->push($index);
+
+            $receivedValues = $this->receiveValue($id, $valuesCollection->getValueConfigs(), $keyStack, $receivedValues);
+
+            $keyStack->pop();
+
+            ++$index;
+        }
+
+        return $receivedValues;
+    }
+
+    /**
+     * @param string $id
+     * @param \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Config\ReceivedValue $valueConfig
+     *
+     * @return mixed
+     */
+    protected function processReceivedValue(string $id, ReceivedValue $valueConfig)
+    {
+        if (!$valueConfig->isRequired() && !$this->needToAskQuestion->ask($id)) {
+            return static::NOT_SET_VALUE;
+        }
+
+        return $this->valueQuestion->ask($valueConfig->getValue());
+    }
+
+    /**
+     * @param string $id
+     * @param \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Config\CallbackValue $valueConfig
+     * @param \SplStack $keyStack
+     * @param array $receivedValues
+     *
+     * @return array
+     */
+    protected function processCallbackValue(string $id, CallbackValue $valueConfig, SplStack $keyStack, array $receivedValues): array
+    {
+        $valueConfig = $valueConfig->getCallback()($receivedValues);
+
+        return $this->receiveValue($id, $valueConfig, $keyStack, $receivedValues);
+    }
+
+    /**
+     * @param array $receivedValues
+     * @param \SplStack $keyStack
+     * @param mixed $value
+     *
+     * @return array
+     */
+    protected function setValueByKeyStack(array $receivedValues, SplStack $keyStack, $value): array
+    {
+        $where = &$receivedValues;
+
+        foreach (array_reverse(iterator_to_array($keyStack)) as $key) {
```

Comment:
what's the magic with links? pls rewrite it

Reply:
Not the magic. This is most optimal way to populate array by sequence of keys placed in another array